From: Jeff Mahoney <jeffm@suse.com>
Subject: btrfs: add cond_resched to btrfs_qgroup_trace_leaf_items
Git-commit: cddf3b2cb33e01087e82580a4a7d508f08ba59e4
Patch-mainline: v4.13-rc1
References: bsc#1028286

On an uncontended system, we can end up hitting soft lockups while
doing replace_path.  At the core, and frequently called is
btrfs_qgroup_trace_leaf_items, so it makes sense to add a cond_resched
there.

Signed-off-by: Jeff Mahoney <jeffm@suse.com>
Reviewed-by: David Sterba <dsterba@suse.com>
Signed-off-by: David Sterba <dsterba@suse.com>
---
 fs/btrfs/qgroup.c |    1 +
 1 file changed, 1 insertion(+)

--- a/fs/btrfs/qgroup.c
+++ b/fs/btrfs/qgroup.c
@@ -1550,6 +1550,7 @@ int btrfs_qgroup_trace_leaf_items(struct
 		if (ret)
 			return ret;
 	}
+	cond_resched();
 	return 0;
 }
 
